projects/plugins/jetpack/modules/woocommerce-analytics/classes/class-jetpack-woocommerce-analytics-checkout-flow.php

```diff
@@ -0,0 +1,213 @@
+				'order_value'      => $order->get_total(),
+				'payment_option'   => $order->get_payment_method(),
+				'products_count'   => $order->get_item_count(),
+				'products'         => $this->format_items_to_json( $order->get_items() ),
+				'order_note'       => $order->get_customer_note(),
+				'shipping_option'  => $order->get_shipping_method(),
+			)
+		);
+	}
+
+	/**
+	 * Track the cart page view
+	 */
+	public function capture_cart_view() {
+		if ( ! is_cart() ) {
+			return;
+		}
+
+		$this->record_event(
+			'woocommerceanalytics_cart_view',
+			array_merge(
+				$this->get_cart_checkout_shared_data(),
+				array()
+			)
+		);
+	}
+
+	/**
+	 * Track the checkout page view
+	 */
+	public function capture_checkout_view() {
+		if ( ! is_checkout() ) {
+			return;
+		}
+
+		// Order received page is also a checkout page, so we need to bail out if we are on that page.
+		if ( is_order_received_page() ) {
+			return;
+		}
+
+		$this->record_event(
+			'woocommerceanalytics_checkout_view',
+			array_merge(
+				$this->get_cart_checkout_shared_data(),
+				array()
+			)
+		);
+	}
+
+	/**
+	 * Format Cart Items or Order Items to an array
+	 *
+	 * @param array|WC_Order_Item[] $items Cart Items or Order Items.
+	 */
+	protected function format_items_to_json( $items ) {
+		$products = array();
+
+		foreach ( $items as $item ) {
+			if ( $item instanceof WC_Order_Item_Product ) {
+				$product = wc_get_product( $item->get_product_id() );
+			} else {
+				$product = $item['data'];
+			}
+			$data = $this->get_product_details( $product );
+			if ( $item instanceof WC_Order_Item_Product ) {
+				$data['pq'] = $item->get_quantity();
+			} else {
+				$data['pq'] = $item['quantity'];
+			}
+			$products[] = $data;
+		}
+
+		return wp_json_encode( $products );
+	}
+
+	/**
+	 * Get Cart/Checkout page view shared data
+	 */
+	protected function get_cart_checkout_shared_data() {
+		$cart = WC()->cart;
+
+		$guest_checkout = ucfirst( get_option( 'woocommerce_enable_guest_checkout', 'No' ) );
+		$create_account = ucfirst( get_option( 'woocommerce_enable_signup_and_login_from_checkout', 'No' ) );
+
+		$coupons     = $cart->get_coupons();
+		$coupon_used = 0;
+		if ( is_countable( $coupons ) ) {
+			$coupon_used = count( $coupons ) ? 1 : 0;
+		}
+
+		$enabled_payment_options = array_filter(
+			WC()->payment_gateways->get_available_payment_gateways(),
+			function ( $payment_gateway ) {
+				if ( ! $payment_gateway instanceof WC_Payment_Gateway ) {
+					return false;
+				}
+
+				return $payment_gateway->is_available();
+			}
+		);
+
+		$enabled_payment_options = array_keys( $enabled_payment_options );
+
+		$shared_data = array(
+			'products'               => $this->format_items_to_json( $cart->get_cart() ),
+			'create_account'         => $create_account,
+			'guest_checkout'         => $guest_checkout,
+			'express_checkout'       => 'null', // TODO: not solved yet.
+			'number_products'        => $cart->get_cart_contents_count(),
```

Comment:
The `woocommerceanalytics_product_checkout` event refers to `number_products`, while the `woocommerceanalytics_checkout_view`, `woocommerceanalytics_order_confirmation_view` and `woocommerceanalytics_product_purchase` call this entry `products_count`. 

To avoid confusion, I wonder if we should also call this entry `products_count` in the `woocommerceanalytics_product_checkout` event.